### ISSUE TITLE
new type inference thinks HTMLCanvasElement is not an EventTarget

### ISSUE THREAD

**Issue by myphysicslab** [google/closure-compiler#832]
Using new type inference compile option --new_type_inf produces this incorrect warning:

```
src/lab/app/SimController.js:255: WARNING - Invalid type for parameter 1 of function goog.events.listen.
expected : EventTarget|goog.events.Listenable|null
found    : HTMLCanvasElement

  this.mouseDownKey_ = goog.events.listen(labCanvas.getCanvas(),
                                          ^
```

HTMLCanvasElement is an Element, which is an EventTarget, so this warning is wrong.

Here is the definition of the `getCanvas` method:

```javascript
/**
* @return {!HTMLCanvasElement}
*/
LabCanvas.prototype.getCanvas = function() {
  return this.canvas_;
};
```

compiler version:
```
Closure Compiler (http://github.com/google/closure-compiler)
Version: v20150126-82-g85da5b6
Built on: 2015/02/11 20:55
```

**Comment by dimvar** [comment]
duplicate of #865

**Comment by myphysicslab** [comment]
confirmed fixed in Version: v20150505-58-g39bb337
